src/CredoCryptoCallback.ts

```diff
@@ -0,0 +1,115 @@
+import { type AgentContext, Buffer, type JwkJson, Key, KeyType, getJwkFromJson, getJwkFromKey } from '@credo-ts/core'
+import { AsnConvert, AsnParser } from '@peculiar/asn1-schema'
+import { type AlgorithmIdentifier, SubjectPublicKeyInfo } from '@peculiar/asn1-x509'
+import { CallbackCryptoKey } from './CallbackKey'
+import type { CryptoCallback } from './CryptoCallback'
+import type {
+  JsonWebKey,
+  KeyAlgorithm,
+  KeyFormat,
+  KeyImportParams,
+  KeySignParams,
+  KeyUsage,
+  KeyVerifyParams,
+} from './types'
+import { ecdsaWithSHA256 } from '@peculiar/asn1-ecc'
+
+export class CredoCryptoCallback implements CryptoCallback<Key> {
```

Comment:
I think this makes sense. With this approach I'm thinking whether we will need an Askar specific crypto implementation at all. If we make the Wallet API in Credo powerfull enough, you wouldn't have to re-implement crypto api for all wallet implemetnation, as it only depends on the wallet API. I think we also might want to move this to Credo core in that case.

I think that is way more powerful. What do you think? Not neccesarily for now. But I think with the refactor you're already a good way in the direction. Do you see any blockers to adopting it? Which methods do we need to add to the wallet api to make it work?

Reply:
Unless we really need an export of the private bytes, I think we can doo everything currently with the wallet API. However... the wallet API can be improved to make it a whole lot easier. Would need to work on it a bit more to get a better overview of the missing methods, but I think I would like an export/import key into the wallet with private parts. (heavily marked as risky and such).

Reply:
Yes something we can tackle with the Wallet API refactor. But aligns well with doing more on top of the wallet-api instead of within so swapping the wallet implementation is easier 👍